src/main/java/com/hubspot/jinjava/lib/filter/AbstractSetFilter.java

```diff
@@ -29,4 +32,62 @@ protected Set<Object> objectToSet(Object var) {
     }
     return result;
   }
+
+  @Override
+  public Object filter(
+    Object var,
+    JinjavaInterpreter interpreter,
+    Object[] args,
+    Map<String, Object> kwargs
+  ) {
+    Set<Object> varSet = objectToSet(var);
+    Set<Object> argSet = objectToSet(parseArgs(interpreter, args));
+
+    attachMismatchedTypesWarning(interpreter, varSet, argSet);
+
+    return filter(varSet, argSet);
+  }
+
+  public abstract Object filter(Set<Object> varSet, Set<Object> argSet);
```

Comment:
Now that the concrete implementations of `AbstractSetFilter` have the `filter` method is wrapped here, all of them will have the mismatched type warning.